spec/expression.dd

```diff
@@ -1100,6 +1100,66 @@ $(GNAME CastExpression):
         (foo) - p;      // subtract p from foo
         -------------
 
+$(H4 $(LNAME2 cast_basic_data_types, Basic Data Types))
+    $(P Cast between different types of data will not be performed automatically.
```

Comment:
This is too vague. One might get the impression that an `uint` must be cast to a `long`, which is not the case. This needs to be explained thoroughly. You can state that where [implicit conversions](https://dlang.org/spec/type.html#implicit-conversions) cannot be applied, casts may be used to force the type system to accept the reinterpretation. The int/byte example below is not necessary as it is thoroughly treated in the above link.

Then  you just need to explain the different types of casts that exist.